Src/AutoFixture.xUnit.net/FreezeAttribute.cs

```diff
@@ -0,0 +1,145 @@
+using System;
+using System.Reflection;
+using Ploeh.AutoFixture.Kernel;
+
+namespace Ploeh.AutoFixture.Xunit
+{
+    /// <summary>
+    /// An attribute that can be applied to parameters in an <see cref="AutoDataAttribute"/>-driven
+    /// Theory to indicate that the parameter value should be frozen and used to satisfy any requests
+    /// made through the same <see cref="IFixture"/> and that match a set of criteria.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false)]
+    public sealed class FreezeAttribute : CustomizeAttribute
```

Comment:
I think you're right when you say that `By` is likely used for interfaces in the vast majority of the cases. On the other hand, if we deprecate `As`, we're left without a way to match a frozen object by a _specific type_ other than the type of the parameter.

That ability would come in handy in situations where a frozen object must be matched by a base type other than the direct parent, i.e. higher up in the inheritance hierarchy.

I recognize that such a necessity is likely a symptom of a problematic design that AutoFixture would _bring attention to_, in case we decided not to address that. However, it might as well be dictated by a pesky framework (I can think of a few) or a legacy system victim of _inheritance frenzy_.

I'm tempted to suggest that we keep `As` around in the `[Frozen]` attribute without obsoleting it and enable the new matching behavior in a new attribute. However, it shouldn't just be called `[Freeze]`, but rather have a more descriptive name like `[FrozenOnMatch]` or something along those lines.